simulators/portal/src/suites/beacon/bridge/service.rs

```diff
@@ -0,0 +1,129 @@
+use crate::suites::beacon::bridge::provider::ConsensusProvider;
+use alloy_primitives::B256;
+use ethportal_api::BeaconContentValue;
+use ethportal_api::BeaconNetworkApiClient;
+use ethportal_api::ContentValue;
+use hivesim::Client;
+use std::sync::Arc;
+use tokio::{
+    sync::Mutex,
+    time::{self, Duration},
+};
+
+// For the sync tests, we need fine-grained control over the data entering
+// the network. The portal-bridge isn't great at providing this flexibility,
+// so instead we have this mock bridge service that we can use to inject
+// data into the network.
+pub struct BridgeService {
+    provider: Arc<ConsensusProvider>,
+    query_interval: Duration,
+    latest_optimistic_root: Arc<Mutex<Option<B256>>>,
+    latest_finalized_root: Arc<Mutex<Option<B256>>>,
+    trusted_block_root: Arc<Mutex<Option<B256>>>,
+    portal_client: Client,
+}
+
+impl BridgeService {
+    pub fn new(portal_client: Client) -> Self {
+        let provider = ConsensusProvider::new().expect("Failed to create consensus provider");
+        Self {
+            provider: Arc::new(provider),
+            query_interval: Duration::from_secs(3),
+            portal_client,
+            latest_optimistic_root: Arc::new(Mutex::new(None)),
+            latest_finalized_root: Arc::new(Mutex::new(None)),
+            trusted_block_root: Arc::new(Mutex::new(None)),
+        }
+    }
+
+    pub async fn latest_optimistic_root(&self) -> Option<B256> {
+        *self.latest_optimistic_root.lock().await
+    }
+
+    pub async fn latest_finalized_root(&self) -> Option<B256> {
+        *self.latest_finalized_root.lock().await
+    }
+
+    pub async fn trusted_block_root(&self) -> Option<B256> {
+        *self.trusted_block_root.lock().await
+    }
+
+    pub async fn start(&self) {
+        let provider = self.provider.clone();
+        let query_interval = self.query_interval;
+        let portal_client = self.portal_client.clone();
+        let trusted_block_root_val = self.trusted_block_root.clone();
+        let finality_update_val = self.latest_finalized_root.clone();
+        let optimistic_update_val = self.latest_optimistic_root.clone();
+
+        tokio::spawn(async move {
+            let mut interval = time::interval(query_interval);
+
+            loop {
+                interval.tick().await;
+
+                // fetch latest finalized root from provider and update the local value
+                let Ok(trusted_block_root) = provider.get_finalized_root().await else {
+                    continue;
+                };
+                {
+                    *trusted_block_root_val.lock().await = Some(trusted_block_root);
+                }
```

Comment:
```suggestion
                *trusted_block_root_val.lock().await = Some(trusted_block_root);
```

We no longer need to scope this to contain the lock